core/src/main/java/org/apache/calcite/sql/SqlBeginEndCall.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+import org.apache.calcite.util.Litmus;
+
+import java.util.List;
+
+import static org.apache.calcite.util.Static.RESOURCE;
+
+/**
+ * Parse tree for {@code SqlBeginEndCall} expression.
+ */
+public class SqlBeginEndCall extends SqlCall {
+  private static final SqlSpecialOperator OPERATOR =
+      new SqlSpecialOperator("BEGIN_END", SqlKind.BEGIN_END);
+
+  public final SqlIdentifier label;
+  public final SqlStatementList statements;
+
+  /**
+   * Creates an instance of {@code SqlBeginEndCall}.
+   *
+   * @param pos SQL parser position
+   * @param beginLabel The label before the BEGIN keyword, may be null
+   * @param endLabel The label after the END keyword, may be null
+   * @param statements List of SQL statements
+   */
+  public SqlBeginEndCall(SqlParserPos pos, SqlIdentifier beginLabel,
+      SqlIdentifier endLabel, SqlStatementList statements) {
+    super(pos);
+    this.label = beginLabel;
+    this.statements = statements;
+    if (endLabel != null && !beginLabel.equalsDeep(endLabel, Litmus.IGNORE)) {
+      throw SqlUtil.newContextException(endLabel.getParserPosition(),
+          RESOURCE.beginEndLabelMismatch());
+    }
+  }
+
+  @Override public void unparse(SqlWriter writer, int leftPrec, int rightPrec) {
+    if (label != null) {
+      label.unparse(writer, 0, 0);
+      writer.setNeedWhitespace(false);
+      writer.print(": ");
+    }
+    writer.keyword("BEGIN");
+    writer.newlineAndIndent();
```

Comment:
How does newlineAndIndent() work? It looks like it indents by SqlPrettyWriter's currentIndent level. And currentIndent level is only modified when startList() is called. Is that correct?

Reply:
Yeah it would indent when the startList is called above. In [SqlDialectParserTest](https://github.com/googleinterns/calcite/blob/master/parsing/src/test/java/org/apache/calcite/test/SqlDialectParserTest.java#L581) the indentation level is currently set to 0 so it doesn't indent and only creates a new line, but if that was set to 4 it would print with spaces correctly in the unit test